## EXP-3
Pranay Tiwari
2019120066

In [ ]:
import seaborn as sns
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import statistics

Importing all necessary libraries to perform SDA

In [ ]:
rd=pd.read_csv("commodity_prices.csv")

In [ ]:
rd.head()

Year  Cocoa  Coffee   Tea  Crude Oil  Coal  Natural Gas  Banana  Sugar  \
0  1960   0.59    0.92  1.03       1.63   NaN         0.14    0.14   0.07   
1  1961   0.48    0.90  0.95       1.57   NaN         0.15    0.13   0.06   
2  1962   0.46    0.83  0.97       1.52   NaN         0.16    0.13   0.06   
3  1963   0.55    0.81  0.95       1.50   NaN         0.16    0.14   0.18   
4  1964   0.51    1.01  0.93       1.45   NaN         0.15    0.17   0.13   

   Orange  ...  Silver  Cotton  Rubber  Tobacco  Coconut Oil  Groundnut Oil  \
0    0.13  ...    0.91    0.65    0.78  1736.87       312.33         327.00   
1    0.13  ...    0.92    0.67    0.60  1563.19       253.42         330.25   
2    0.11  ...    1.08    0.65    0.56  1404.90       251.22         274.34   
3    0.09  ...    1.28    0.65    0.52  1343.61       286.33         268.50   
4    0.08  ...    1.29    0.65    0.49  1488.30       296.48         315.38   

   Palm Oil  Soybean   Logs  Sawnwood  
0    224.42   223.92  31.94    149.17  
1    228.42   287.33  33.06    151.98  
2    216.24   227.00  36.39    160.06  
3    222.39   223.58  35.49    157.85  
4    239.45   233.00  30.26    144.83  

[5 rows x 31 columns]

In [ ]:
rd.mean()

Year             1990.500000
Cocoa               1.719677
Coffee              2.527097
Tea                 1.752581
Crude Oil          30.891452
Coal               48.505769
Natural Gas         2.336935
Banana              0.500323
Sugar               0.237097
Orange              0.475645
Barley             83.452131
Maize             120.307097
Sorghum           113.635574
Rice              296.025323
Wheat             154.524839
Beef                2.413548
Chicken             1.136935
Lamb                3.941765
Shrimps             9.493871
Gold              510.236613
Platinum          566.927903
Silver              8.678226
Cotton              1.433710
Rubber              1.218387
Tobacco          2809.714194
Coconut Oil       674.460806
Groundnut Oil     938.567097
Palm Oil          512.494032
Soybean           574.069677
Logs              175.032097
Sawnwood          489.774355
dtype: float64

The avg commodity price of tobacco is 2809.714. The avg price of Sugar is 0.23709677419354838 which is the lowest.

In [ ]:
t = max(rd.mean())
print(t)
f = min(rd.mean())
f

2809.714193548386


0.23709677419354838

In [ ]:
rd.median()

Year             1990.500
Cocoa               1.605
Coffee              2.705
Tea                 1.655
Crude Oil          19.895
Coal               38.130
Natural Gas         1.995
Banana              0.435
Sugar               0.210
Orange              0.440
Barley             77.230
Maize             109.270
Sorghum           105.110
Rice              274.975
Wheat             149.395
Beef                2.300
Chicken             1.070
Lamb                2.880
Shrimps            10.785
Gold              362.845
Platinum          423.825
Silver              5.350
Cotton              1.450
Rubber              0.965
Tobacco          2717.780
Coconut Oil       584.170
Groundnut Oil     891.750
Palm Oil          487.990
Soybean           548.045
Logs              187.135
Sawnwood          484.715
dtype: float64

The median values of orange, beef, cotton are as same as their mean values, hence their average commodity prices are same as their middle price values, hence the dataspread is not varied.

In [ ]:
rd.mode()

Year  Cocoa  Coffee   Tea  Crude Oil   Coal  Natural Gas  Banana  Sugar  \
0   1960   2.89    1.37  1.66       1.21  39.67         0.16    0.14   0.28   
1   1961    NaN    3.08   NaN       1.27    NaN          NaN    0.37    NaN   
2   1962    NaN    3.32   NaN       1.32    NaN          NaN     NaN    NaN   
3   1963    NaN     NaN   NaN       1.33    NaN          NaN     NaN    NaN   
4   1964    NaN     NaN   NaN       1.36    NaN          NaN     NaN    NaN   
..   ...    ...     ...   ...        ...    ...          ...     ...    ...   
57  2017    NaN     NaN   NaN      96.24    NaN          NaN     NaN    NaN   
58  2018    NaN     NaN   NaN      96.99    NaN          NaN     NaN    NaN   
59  2019    NaN     NaN   NaN     104.01    NaN          NaN     NaN    NaN   
60  2020    NaN     NaN   NaN     104.08    NaN          NaN     NaN    NaN   
61  2021    NaN     NaN   NaN     105.01    NaN          NaN     NaN    NaN   

    Orange  ...  Silver  Cotton  Rubber  Tobacco  Coconut Oil  Groundnut Oil  \
0     0.17  ...    1.29    0.65    0.86  1042.65       234.06         268.50   
1     0.40  ...     NaN     NaN     NaN  1073.64       251.22         270.73   
2      NaN  ...     NaN     NaN     NaN  1075.98       253.42         274.34   
3      NaN  ...     NaN     NaN     NaN  1112.38       286.33         282.95   
4      NaN  ...     NaN     NaN     NaN  1128.02       296.48         296.31   
..     ...  ...     ...     ...     ...      ...          ...            ...   
57     NaN  ...     NaN     NaN     NaN  4627.36      1281.12        1820.62   
58     NaN  ...     NaN     NaN     NaN  4790.50      1481.78        1866.09   
59     NaN  ...     NaN     NaN     NaN  4865.96      1636.31        2075.09   
60     NaN  ...     NaN     NaN     NaN  4908.30      1650.65        2144.74   
61     NaN  ...     NaN     NaN     NaN  4990.77      1730.08        2359.06   

    Palm Oil  Soybean    Logs  Sawnwood  
0     168.83   178.08   30.26    144.83  
1     181.11   197.17   31.94    149.17  
2     216.24   216.83   33.06    151.98  
3     217.26   223.58   35.03    156.75  
4     222.39   223.92   35.49    157.85  
..       ...      ...     ...       ...  
57    933.02  1055.39  305.43    852.80  
58   1042.92  1225.60  307.55    876.34  
59   1043.40  1260.81  360.51    889.09  
60   1130.58  1297.66  389.81    897.94  
61   1193.37  1385.41  390.51    939.45  

[62 rows x 31 columns]

Mode of all prices of all commodities

In [ ]:
rd.skew()

Year             0.000000
Cocoa            0.320483
Coffee           0.353664
Tea              0.341637
Crude Oil        1.157491
Coal             1.253486
Natural Gas      1.309507
Banana           0.775601
Sugar            0.942986
Orange           0.431370
Barley           1.087130
Maize            1.212836
Sorghum          1.036623
Rice             0.611817
Wheat            0.712940
Beef             0.707343
Chicken          0.302458
Lamb             1.177052
Shrimps         -0.681148
Gold             1.226473
Platinum         0.989621
Silver           1.444919
Cotton           0.434822
Rubber           2.001157
Tobacco          0.237457
Coconut Oil      1.185483
Groundnut Oil    0.755748
Palm Oil         0.799648
Soybean          0.938917
Logs             0.088749
Sawnwood         0.101718
dtype: float64

Skewness characterizes the degree of asymmetry of a distribution around its mean

Positive skewness indicates a distribution with an asymmetric tail extending toward more positive values

Negative skewness indicates a distribution with an asymmetric tail extending toward more negative values

The skewness of Logs, Sawnwood, Tobacco, Shrimps, Chicken, Orange, Tea, Coffee, Cocoa are closest to zero indicating  that these attributes have similar commodity prices in the dataset.

In [ ]:
statistics.variance(rd['Orange'])

0.07912662612374406

Variance value for the attribute: orange is very low which indicates that there is less deviation from the mean of the attribute.

In [ ]:
#z test
from numpy.random import randn
from statsmodels.stats.weightstats import ztest

alpha=0.05
null_hypo=0.500323

ztest_Score, p_value= ztest(rd['Banana'],value = null_hypo, alternative='larger')
# the function outputs a p_value and z-score corresponding to that value, we compare the 
# p-value with alpha, if it is greater than alpha then we do not null hypothesis 
# else we reject it.
  
if(p_value <  alpha):
  print("Reject Null Hypothesis")
else:
  print("Fail to Reject NUll Hypothesis thus null hypothesis true")

Fail to Reject NUll Hypothesis thus null hypothesis true


Z-test is a statistical method to determine whether the distribution of the test statistics can be approximated by a normal distribution

It is the method to determine whether two sample means are approximately the same or different when their variance is known and the sample size is large